eyelid-match-ops/src/primitives/yashe.rs

```diff
@@ -0,0 +1,119 @@
+//! Implementation of YASHE cryptosystem
+//! `<https://eprint.iacr.org/2013/075.pdf>`
+
+use crate::primitives::poly::{inverse, modular_poly::Poly, Coeff};
+use ark_ff::{One, UniformRand, Zero};
+use rand::rngs::ThreadRng;
+use rand_distr::{Distribution, Normal};
+
+pub mod test;
+
+/// Yashe parameters
+#[derive(Clone)]
+pub struct YasheParams {
+    /// Plaintext coefficient modulus
+    pub t: u64,
```

Comment:
Should the type of this parameter be `Coeff` instead?
If we use `u64`, then the top 13 of the 77 bits can't be set (which also means we can't have small negative values).

Are there any security constraints on its value?
(For example, can it be zero or one?)

Reply:
This parameter must be a positive integer greater than or equal to 2. It is the modulus of plaintext coefficients. For our instantiations, it will use at most 25 bits.

Reply:
Thanks, can you document that, and assert that it always holds? (if it's easy to add an assert)

I can deal with the types and any changes as part of #59.

Reply:
I added the documentation task to ticket #70